pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -749,68 +755,94 @@ func (og *operationGenerator) GenerateVerifyControllerAttachedVolumeFunc(
 			return nil
 		}
 
-		if !volumeToMount.ReportedInUse {
-			// If the given volume has not yet been added to the list of
-			// VolumesInUse in the node's volume status, do not proceed, return
-			// error. Caller will log and retry. The node status is updated
-			// periodically by kubelet, so it may take as much as 10 seconds
-			// before this clears.
-			// Issue #28141 to enable on demand status updates.
-			return fmt.Errorf("Volume %q (spec.Name: %q) pod %q (UID: %q) has not yet been added to the list of VolumesInUse in the node's volume status",
-				volumeToMount.VolumeName,
-				volumeToMount.VolumeSpec.Name(),
-				volumeToMount.PodName,
-				volumeToMount.Pod.UID)
-		}
-
-		// Fetch current node object
-		node, fetchErr := og.kubeClient.Core().Nodes().Get(string(nodeName), metav1.GetOptions{})
-		if fetchErr != nil {
+		timeout := int64(nodeWatchTimeoutInSeconds)
+		nodeSelector := fields.OneTermEqualSelector("metadata.name", string(nodeName))
+		w, err := og.kubeClient.Core().Nodes().Watch(metav1.ListOptions{
+			FieldSelector:   nodeSelector.String(),
+			ResourceVersion: "0",
+			TimeoutSeconds:  &timeout,
+		})
+		if err != nil {
 			// On failure, return error. Caller will log and retry.
 			return fmt.Errorf(
-				"VerifyControllerAttachedVolume failed fetching node from API server. Volume %q (spec.Name: %q) pod %q (UID: %q). Error: %v",
+				"VerifyControllerAttachedVolume failed watching node on API server. Volume %q (spec.Name: %q) pod %q (UID: %q). Error: %v",
 				volumeToMount.VolumeName,
 				volumeToMount.VolumeSpec.Name(),
 				volumeToMount.PodName,
 				volumeToMount.Pod.UID,
-				fetchErr)
+				err)
 		}
 
-		if node == nil {
-			// On failure, return error. Caller will log and retry.
-			return fmt.Errorf(
-				"VerifyControllerAttachedVolume failed. Volume %q (spec.Name: %q) pod %q (UID: %q). Error: node object retrieved from API server is nil",
-				volumeToMount.VolumeName,
-				volumeToMount.VolumeSpec.Name(),
-				volumeToMount.PodName,
-				volumeToMount.Pod.UID)
-		}
+		volumeInUse := false
 
-		for _, attachedVolume := range node.Status.VolumesAttached {
-			if attachedVolume.Name == volumeToMount.VolumeName {
-				addVolumeNodeErr := actualStateOfWorld.MarkVolumeAsAttached(
-					v1.UniqueVolumeName(""), volumeToMount.VolumeSpec, nodeName, attachedVolume.DevicePath)
-				glog.Infof("Controller successfully attached volume %q (spec.Name: %q) pod %q (UID: %q) devicePath: %q",
+		for r := range w.ResultChan() {
```

Comment:
I'm not sure if I understand you correctly. I used https://gobyexample.com/range-over-channels as an example on how to iterate the messages from a channel. I'm not a Go pro, so I'm sorry if I misunderstood something here.

Or do you mean the case were the same node is waiting for multiple volumes to be attached? In that case, yes, there would be multiple watches/channels for this node. But I expected this to be ok as it does not happen all the time and is only a temporary situation.